### PR TITLE
en_dy_extra OK

### PR DESCRIPTION
**Pull request by fullgrowngaming**

### CODE REVIEW

File: src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c [hunk 1]
```diff
@@ -21,16 +23,104 @@ const ActorInit En_Dy_Extra_InitVars = {
...
+    this->unk160 = this->actor.currPosRot.pos;
+    this->actor.gravity = -0.2f;
+    this->unk150 = 1.0f;
+    this->unk14C = 0x3C;
+    this->actionFunc = func_80A61334;
+}
+
+void func_80A61334(EnDyExtra* this, GlobalContext* globalCtx) {
+    Math_SmoothScaleMaxF(&this->actor.gravity, 0.0f, 0.1f, 0.005f);
+    if (this->actor.currPosRot.pos.y < -85.0f) {
```
Comment (m4xw) [inline src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c:49]: \n

File: src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c [hunk 2]
```diff
@@ -21,16 +23,104 @@ const ActorInit En_Dy_Extra_InitVars = {
...
+        this->actor.velocity.y = 0.0f;
+    }
+    if (this->unk14C == 0 && this->unk14A != 0) {
```
Comment (m4xw) [inline src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c:52]: \n before as above

File: src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c [hunk 3]
```diff
@@ -21,16 +23,104 @@ const ActorInit En_Dy_Extra_InitVars = {
...
+    }
+}
+
+void func_80A613C8(EnDyExtra* this, GlobalContext* globalCtx) {
+    Math_SmoothScaleMaxF(&this->actor.gravity, 0.0f, 0.1f, 0.005f);
+    if (this->unk14C == 0 || this->unk150 < 0.02f) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+    this->unk150 -= 0.02f;
```
Comment (m4xw) [inline src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c:64]: \n before this

File: src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c [hunk 4]
```diff
@@ -21,16 +23,104 @@ const ActorInit En_Dy_Extra_InitVars = {
...
+    }
+    if (this->unk14C == 0 && this->unk14A != 0) {
+        this->unk14C = 0x32;
+        this->actionFunc = func_80A613C8;
+    }
+}
+
+void func_80A613C8(EnDyExtra* this, GlobalContext* globalCtx) {
+    Math_SmoothScaleMaxF(&this->actor.gravity, 0.0f, 0.1f, 0.005f);
+    if (this->unk14C == 0 || this->unk150 < 0.02f) {
```
Comment (m4xw) [inline src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c:60]: \n before this

File: include/color.h [hunk 5]
```diff
@@ -23,4 +23,8 @@ typedef struct {
     /* 0x3 */ u8 alpha;
 } RGBA8; // size = 0x4
 
+typedef struct {
+    u8 r, g, b, a;
+} Color_RGBA8_n;
```
Comment (m4xw) [inline include/color.h:28]: as discussed

**Review by m4xw — CHANGES_REQUESTED** [review]

File: src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c (hunk 1 above)
Comment (m4xw) [reply src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c:49]: (before this)

**Review by Kenix3 — APPROVED** [review]
Looks good to me.
Build is OK.

**Comment by fullgrowngaming** [comment]
Fixed the unnecessary color thing, but I'm a bit confused about the newline suggestions

**Comment by fig02** [comment]
I prefer separating out blocks myself like m4 suggested, but it might be a bit cumbersome to enforce project wide. at least in oot we dont enforce anything related to that, dunno if mm agreed on a style for that tho

**Comment by fullgrowngaming** [comment]
(went ahead and added the newlines, but yeah, we should probably discuss whether that's something to enforce at this stage)

**Review by Rozelette — APPROVED** [review]
LGTM

**Pull request merged**